### ISSUE TITLE
TypeError in data_sets API

### ISSUE THREAD

**Issue by hackdna** [refinery-platform/refinery-platform#3251]
* Specific code commit: v1.6.8
* Version of the web browser and OS: N/A
* Environment where the error occurred: AWS

### Steps to reproduce
Unknown, error log received by email.

### Observed behavior
```
Internal Server Error: /api/v2/data_sets/26f22113-3b06-4a64-9019-d400a05ffcb3/
Traceback (most recent call last):
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/rest_framework/views.py", line 466, in dispatch
    response = self.handle_exception(exc)
  File "/home/ubuntu/.virtualenvs/refinery-platform/lib/python2.7/site-packages/rest_framework/views.py", line 463, in dispatch
    response = handler(request, *args, **kwargs)
TypeError: get() got an unexpected keyword argument 'uuid'

Request repr():
<WSGIRequest
path:/api/v2/data_sets/26f22113-3b06-4a64-9019-d400a05ffcb3/,
GET:<QueryDict: {}>,
POST:<QueryDict: {}>,
COOKIES:{'_ga': 'GA1.2.1639004996.1517425836',
'_gid': 'GA1.2.405133548.1552312341',
'csrftoken': '<redacted>',
'sessionid': '<redacted>'},
META:{'CONTEXT_DOCUMENT_ROOT': '/var/www',
'CONTEXT_PREFIX': '',
u'CSRF_COOKIE': u'<redacted>',
'DOCUMENT_ROOT': '/var/www',
'GATEWAY_INTERFACE': 'CGI/1.1',
'HTTP_ACCEPT': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8',
'HTTP_ACCEPT_ENCODING': 'gzip, deflate, br',
'HTTP_ACCEPT_LANGUAGE': 'en-US,en;q=0.9',
'HTTP_CONNECTION': 'keep-alive',
'HTTP_COOKIE': '_ga=GA1.2.1639004996.1517425836; csrftoken=<redacted>; sessionid=<redacted>; _gid=GA1.2.405133548.1552312341',
'HTTP_HOST': 'hsci.stemcellcommons.org',
'HTTP_UPGRADE_INSECURE_REQUESTS': '1',
'HTTP_USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/72.0.3626.121 Safari/537.36',
'HTTP_X_FORWARDED_FOR': '134.174.140.132',
'HTTP_X_FORWARDED_PORT': '443',
'HTTP_X_FORWARDED_PROTO': 'https',
'PATH_INFO': u'/api/v2/data_sets/26f22113-3b06-4a64-9019-d400a05ffcb3/',
'PATH_TRANSLATED': '/srv/refinery-platform/refinery/config/wsgi_aws.py/api/v2/data_sets/26f22113-3b06-4a64-9019-d400a05ffcb3/',
'QUERY_STRING': '',
'REMOTE_ADDR': '172.30.0.16',
'REMOTE_PORT': '30856',
'REQUEST_METHOD': 'GET',
'REQUEST_SCHEME': 'http',
'REQUEST_URI': '/api/v2/data_sets/26f22113-3b06-4a64-9019-d400a05ffcb3/',
'SCRIPT_FILENAME': '/srv/refinery-platform/refinery/config/wsgi_aws.py',
'SCRIPT_NAME': u'',
'SCRIPT_URI': 'http://hsci.stemcellcommons.org/api/v2/data_sets/26f22113-3b06-4a64-9019-d400a05ffcb3/',
'SCRIPT_URL': '/api/v2/data_sets/26f22113-3b06-4a64-9019-d400a05ffcb3/',
'SERVER_ADDR': '172.30.0.74',
'SERVER_ADMIN': '[no address given]',
'SERVER_NAME': 'hsci.stemcellcommons.org',
'SERVER_PORT': '80',
'SERVER_PROTOCOL': 'HTTP/1.1',
'SERVER_SIGNATURE': '',
'SERVER_SOFTWARE': 'Apache/2.4.7 (Ubuntu)',
'mod_wsgi.application_group': 'hsci.stemcellcommons.org|',
'mod_wsgi.callable_object': 'application',
'mod_wsgi.enable_sendfile': '0',
'mod_wsgi.handler_script': '',
'mod_wsgi.input_chunked': '0',
'mod_wsgi.listener_host': '',
'mod_wsgi.listener_port': '80',
'mod_wsgi.process_group': 'refinery',
'mod_wsgi.queue_start': '1552317358730233',
'mod_wsgi.request_handler': 'wsgi-script',
'mod_wsgi.script_reloading': '1',
'mod_wsgi.version': (3, 4),
'wsgi.errors': <mod_wsgi.Log object at 0x7feed92d9e30>,
'wsgi.file_wrapper': <built-in method file_wrapper of mod_wsgi.Adapter object at 0x7feed8d94eb8>,
'wsgi.input': <mod_wsgi.Input object at 0x7feed8fac7f0>,
'wsgi.multiprocess': False,
'wsgi.multithread': True,
'wsgi.run_once': False,
'wsgi.url_scheme': 'http',
'wsgi.version': (1, 0)}>
```

### Expected behavior
No internal server error

**Comment by jkmarx** [comment]
@gmnelson @hackdna The extra data_set v2 api error calls are from me (I was trying to sort out the issue).  I believed I cause the original issue when I was trying to ping the v1 api yesterday. The current version of the HSCI site data set v2 api does not support data set details, but the error was not handled correctly. It will be resolved in the new release, as I am in the process of extending the data set api for details. https://github.com/refinery-platform/refinery-platform/issues/3251

**Comment by hackdna** [comment]
Is there a way to configure DRF so that APIs don't die by default when an extra keyword argument is used?

**Comment by jkmarx** [comment]
@hackdna I'll look into it.

**Comment by jkmarx** [comment]
Resolved by #3252 
Extended the API to retrieve data sets by UUID
